Shared/mods/deathmatch/logic/luadefs/CLuaElementDefsShared.cpp

```diff
@@ -1,87 +1,78 @@
 #include "StdInc.h"
 
-int CLuaElementDefs::GetElementData(lua_State* luaVM)
-{
-    //  var getElementData ( element theElement, string key [, inherit = true] )
+// Warn and truncate if key is too long(luaVM must be in a var luaVM)
```

Comment:
I'd prefer this to be a (private static) function rather than a macro

Reply:
This. C++ doesn't like macros.

Reply:
yeah, but then i need to pass in m_pScriptDebugging, which seemed a bit odd. Or i could define it a static CLuaElemetnDefs function, which would be confusing.
Which is the better way?

Reply:
>  Or i could define it a static CLuaElemetnDefs function, which would be confusing.

I'd say it's better to make it a static function outside of CLuaElementDefs::GetElementData class. This function will be inaccessible to other .cpp files, and that's a good thing. By design, `CLuaElementDefs` is meant to have Lua functions, adding utility functions just doesn't look right. There are other ways to properly refactor the code, but this is a simple solution, and I think it's good.

Reply:
You mean adding a function to `CLuaElementDefsShared.cpp` like, 
```cpp
void TruncateStringAndWarn(std::string& key, lua_State* const luaVM, CBlah const scriptDebugging);
```

Reply:
Yes, and don't forget to add the `static` keyboard before the function.  :)

```
static void TruncateStringAndWarn(std::string& key, lua_State* const luaVM, CBlah const scriptDebugging)
{
// code
}
```

Reply:
Yeah, thanks for reminding!